variables.tf

```diff
@@ -2,8 +2,106 @@
 # Input Variables
 ########################################################################################################################
 
-#variable "my_variable" {
+variable "resource_group_id" {
+  type        = string
+  description = "The ID of the resource group to provision the MQ on Cloud instance to."
+}
+
+variable "region" {
+  type        = string
+  description = "The region to provision the MQ on Cloud instance to."
+}
+
+variable "name" {
+  type        = string
+  description = "The name to give the MQ on Cloud instance."
+}
+
+#variable "service_plan" {
 #  type        = string
-#  description = "A description of my variable"
-#  default     = "default_value"
+#  description = "The MQ on Cloud plan to provision."
+#  default     = "xsmall"
 #}
+
+variable "tags" {
+  type        = list(string)
+  description = "The list of resource tags that you want to associate with your MQ on Cloud instance."
+  default     = []
+}
+
+variable "existing_mq_capacity_guid" {
+  type        = string
+  description = "The GUID of an existing capacity service instance, if not specifed, a new capacity plan will be created"
+  default     = null
+}
+
+########################################################################################################################
+# Queue Manager Variables
+########################################################################################################################
+
+variable "queue_manager_display_name" {
+  description = "A displayable name for the queue manager."
+  type        = string
+}
+
+variable "queue_manager_location" {
+  description = "The location in which the queue manager will be deployed. Defaults to using the first location in the created service instance"
+  type        = string
+  default     = null
+}
+
+variable "queue_manager_name" {
+  type        = string
+  description = "A queue manager name conforming to MQ restrictions. 1 to 48 characters matching regular expression '/^[a-zA-Z0-9._]*$/' ."
+}
+
+variable "queue_manager_size" {
+  description = "The queue manager deployment sizes. Valid values are `xsmall`, `small`, `medium`, `large` ."
+  type        = string
+  default     = "xsmall"
+}
+
+variable "queue_manager_version" {
+  description = "The MQ version of the queue manager."
+  type        = string
+  default     = "9.3.2_2"
```

Comment:
How often are versions released? Should we only support certain versions that we have tested?
We will want to decide how to manage this version (e.g. can we create a custom renovate rule, or some other automation to create a PR)

Reply:
There are one or two versions a month (usually). Only two versions available at any time. Bigger problem potentially they are aggressively updated, so any version actually deployed may be updated under the covers. Backward `upgrades` are not permitted; so terraform can get stale.

I am not sure that renovate will work. This is a property of the deployment instance, rather than the service. I expect there are times when different instances may support different values.

Longer term, I am inclined to make this required. The tests can find the value and run with the `latest` value. The demo example does this.